internal/services/containers/migration/kubernetes_cluster.go

```diff
@@ -0,0 +1,1421 @@
+package migration
+
+import (
+	"context"
+	"log"
+
+	"github.com/Azure/azure-sdk-for-go/services/containerservice/mgmt/2021-08-01/containerservice"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/location"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/containers/parse"
+	containerValidate "github.com/hashicorp/terraform-provider-azurerm/internal/services/containers/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tags"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/suppress"
+)
+
+var _ pluginsdk.StateUpgrade = KubernetesClusterV0ToV1{}
+
+type KubernetesClusterV0ToV1 struct{}
+
+func (k KubernetesClusterV0ToV1) UpgradeFunc() pluginsdk.StateUpgraderFunc {
+	return func(ctx context.Context, rawState map[string]interface{}, meta interface{}) (map[string]interface{}, error) {
+		log.Printf("[DEBUG] Migrating ID to correct casing for Kubernetes Cluster")
+		rawId := rawState["id"].(string)
+
+		id, err := parse.ClusterID(rawId)
+		if err != nil {
+			return nil, err
+		}
+
+		rawState["id"] = id.ID()
+		return rawState, nil
+	}
+}
+
+func (k KubernetesClusterV0ToV1) Schema() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+		},
+
+		"location": {
+			Type:             schema.TypeString,
+			Required:         true,
+			ForceNew:         true,
+			StateFunc:        location.StateFunc,
+			DiffSuppressFunc: location.DiffSuppressFunc,
```

Comment:
State migrations only require the Type and Required/Optional/Computed for most fields, things like List of Strings also require the Elem Type, so these fields can be removed everywhere in `Schema()`

Reply:
Copy that, I'll remove these functions, thanks for the tip!